aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1851,6 +1813,74 @@ static Tensor& linalg_norm_out_impl(Tensor& result, const Tensor& self, optional
   return result;
 }
 
+static Tensor& linalg_vector_norm_impl(const Tensor& self, optional<Scalar> opt_ord, optional<IntArrayRef> opt_dim, bool keepdim, optional<ScalarType> opt_dtype, Tensor& result) {
+  auto ord = opt_ord.value_or(2).toDouble();
+
+  TORCH_CHECK(self.device().type() == DeviceType::CPU || self.device().type() == DeviceType::CUDA,
+              "vector_norm only supports CPU and CUDA device types, but got: ", self.device().type());
+  TORCH_CHECK(self.layout() == Layout::Strided,
+              "vector_norm only supports strided layout, but got: ", self.layout());
+
+  ScalarType in_dtype = opt_dtype.has_value() ? opt_dtype.value() : self.scalar_type();
+  TORCH_CHECK(
+      at::isFloatingType(in_dtype) || at::isComplexType(in_dtype),
+      "vector_norm only supports floating point and complex dtypes, but got: ",
+      toString(in_dtype));
+
+  IntArrayRef dim = opt_dim.value_or(IntArrayRef{});
+
+  // For NumPy compatibility, we can only perform order +/-infinity reduction
```

Comment:
Per offline discussion, let's be very safe in this PR and throw an error for now when dealing with no elements and a negative ord to address https://github.com/pytorch/pytorch/issues/52783. That way we can change the behavior later (if we want) without making a BC-breaking change.

We'll need to add a test for this, and it's not NumPy-compatible, but that's OK.

Reply:
I've added an error that triggers when the input is empty and ord is negative